core/src/main/java/tachyon/client/RemoteBlockReader.java

```diff
@@ -4,20 +4,36 @@
  * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance with the License. You may obtain a
  * copy of the License at
- * 
  * http://www.apache.org/licenses/LICENSE-2.0
- * 
  * Unless required by applicable law or agreed to in writing, software distributed under the License
  * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
  * or implied. See the License for the specific language governing permissions and limitations under
  * the License.
  */
 
-package tachyon.worker;
+package tachyon.client;
 
-/**
- * Enum over the different networking implementations.
- */
-public enum NetworkType {
-  NIO, NETTY
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import com.google.common.base.Throwables;
+
+import tachyon.conf.UserConf;
+import tachyon.util.CommonUtils;
+
+public interface RemoteBlockReader {
+
+  class Factory {
```

Comment:
Do you mean move the factory to a new class named RemoteBlockReaders or change the interface name?

I preferred to combine the factory and the interface since neither one has much substance by it's own.